### ISSUE TITLE
[Bug][a11y]:  Picker: VoiceOver fails to announce label in Chrome

### ISSUE THREAD

**Issue by majornista** [adobe/spectrum-web-components#3238]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

Picker

### Expected behavior

The button element in the sp-picker component's shadow DOM is labeled internally using aria-labelledby:
1. by the button itself, which has an aria-label that comes from the textContent of the associated sp-field-label component,
2. by the icon slot within the shadow DOM,
3. and by the label slot within the shadow DOM, that will contain either the selected value or the assigned placeholder text.

VoiceOver should announce the menu button using the accessibility name assembled from elements referenced in the aria-labelledby attribute, but instead it only announces the component description, "menu popup, collapsed button."

### Actual behavior

Steps to reproduce the problem:
1. With VoiceOver on, open https://opensource.adobe.com/spectrum-web-components/components/picker/#sizes.
2. Navigate to focus the Dropdown menu button, or Picker, labelled "Selection type:" with a placeholder value "Selection type" under the heading Sizes and within the Medium tab panel.
3. VoiceOver announces the Dropdown menu button as "menu popup, collapsed button," without including the accessibility name, which, if you look at the Accessibility properties panel in the Dev tools, does seem to be properly assembled and communicated as "Selection type: Selection type." 

The issue does not reproduced using Safari or Webkit.

### Related Chromium bug

[Issue 1448092: VoiceOver fails to announce label for web component menu button](https://bugs.chromium.org/p/chromium/issues/detail?id=1448092)

### What browsers are you seeing the problem in?

Chrome, Microsoft Edge

### How can we reproduce this issue?

1. With VoiceOver on, open https://opensource.adobe.com/spectrum-web-components/components/picker/#sizes.
2. Navigate to focus the Dropdown menu button, or Picker, labelled "Selection type:" with a placeholder value "Selection type" under the heading Sizes and within the Medium tab panel.
3. VoiceOver announces the Dropdown menu button as "menu popup, collapsed button," without including the accessibility name, which, if you look at the Accessibility properties panel in the Dev tools, does seem to be properly assembled and communicated as "Selection type: Selection type." 

The issue does not reproduced using Safari or Webkit.

### Sample code that illustrates the problem

See CodePen example: https://codepen.io/majornista/pen/BaqMQJg
